src/mpid/ch4/shm/xpmem/xpmem_send.h

```diff
@@ -0,0 +1,78 @@
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil ; -*- */
+/*
+ *  (C) 2006 by Argonne National Laboratory.
+ *      See COPYRIGHT in top-level directory.
+ *
+ *  Portions of this code were written by Intel Corporation.
+ *  Copyright (C) 2011-2016 Intel Corporation.  Intel provides this material
+ *  to Argonne National Laboratory subject to Software Grant and Corporate
+ *  Contributor License Agreement dated February 8, 2012.
+ */
+
+#ifndef XPMEM_SEND_H_INCLUDED
+#define XPMEM_SEND_H_INCLUDED
+
+#include "ch4_impl.h"
+#include "xpmem_control.h"
+#include "xpmem_pre.h"
+
+#undef FUNCNAME
+#define FUNCNAME MPIDI_XPMEM_lmt_isend
+#undef FCNAME
+#define FCNAME MPL_QUOTE(FUNCNAME)
+MPL_STATIC_INLINE_PREFIX int MPIDI_XPMEM_lmt_isend(const void *buf, MPI_Aint count,
+                                                   MPI_Datatype datatype, int rank, int tag,
+                                                   MPIR_Comm * comm, int context_offset,
+                                                   MPIDI_av_entry_t * addr, MPIR_Request ** request)
+{
+    int mpi_errno = MPI_SUCCESS;
+    MPIR_Request *sreq = NULL;
+    size_t data_sz;
+    MPI_Aint true_lb;
+    bool is_contig = 0;
```

Comment:
```suggestion
    bool is_contig ATTRIBUTE((unused)) = 0;
```

If you don't have assertions turned on, this will produce a ton of warnings.

Reply:
Added